builtins/amp-img.js

```diff
@@ -89,6 +84,14 @@ export class AmpImg extends BaseElement {
     return promise;
   }
 
+  /** @override */
+  layoutCallback() {
+    // This is specifically needed for elements that does its work on prerender
+    // e.g. amp-img. This is only needed because currently prerendering
+    // undisplayed elements does not work. See issue #2742.
+    return this.updateImageSrc_();
```

Comment:
In the case described above, where we'd potentially use a lower-quality image for prerendering, this is an obvious risk condition where an lower quality image can overwrite the higher quality image.